File: src/ros2_bridge/mappers/foxglove_msgs/packed_element_field.rs
//! Typed mapper for `foxglove_msgs/msg/PackedElementField`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn packed_element_field_to_bus(msg: ros_env::foxglove_msgs::msg::PackedElementField) -> crate::foxglove_msgs::msg::v1::PackedElementField {
    crate::foxglove_msgs::msg::v1::PackedElementField {
        name: crate::ros2_bridge::mappers::convert::from_ros_string(msg.name),
        offset: msg.offset,
        r#type: msg.type_ as i32,
    }
}

pub(crate) fn packed_element_field_to_ros(bus: crate::foxglove_msgs::msg::v1::PackedElementField) -> ros_env::foxglove_msgs::msg::PackedElementField {
    ros_env::foxglove_msgs::msg::PackedElementField {
        name: crate::ros2_bridge::mappers::convert::to_ros_string(bus.name),
        offset: bus.offset,
        type_: bus.r#type as i32,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct FoxgloveMsgsPackedElementFieldMapper;

impl TypedTopicMapper for FoxgloveMsgsPackedElementFieldMapper {
    type Ros = ros_env::foxglove_msgs::msg::PackedElementField;
    type Bus = crate::foxglove_msgs::msg::v1::PackedElementField;

    fn type_name(&self) -> &'static str {
        "foxglove_msgs/msg/PackedElementField"
    }

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(packed_element_field_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(packed_element_field_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/foxglove_msgs/packed_element_field.py
"""Generated mapper for `foxglove_msgs/msg/PackedElementField`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.foxglove_msgs.msg.v1 import PackedElementField as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def packed_element_field_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.name = str(msg.name)
    bus.offset = msg.offset
    bus.type = int(msg.type)
    return bus


def packed_element_field_to_ros(bus):
    from foxglove_msgs.msg import PackedElementField as RosMsg

    out = RosMsg()
    out.name = str(bus.name)
    out.offset = bus.offset
    out.type = int(bus.type)
    return out


class FoxgloveMsgsPackedElementFieldMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from foxglove_msgs.msg import PackedElementField as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return packed_element_field_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return packed_element_field_to_ros(bus)
